prov/sockets/src/sock_fabric.c

```diff
@@ -509,6 +514,38 @@ static int sock_ep_getinfo(const char *node, const char *service, uint64_t flags
 	return ret;
 }
 
+static void getnodename(char *buf, int buflen)
+{
+	int ret;
+	struct addrinfo ai, *rai = NULL;
+	struct ifaddrs *ifaddrs, *ifa;
+
+	gethostname(buf, buflen);
+	ret = getaddrinfo(buf, NULL, &ai, &rai);
+	if (!ret) {
+		freeaddrinfo(rai);
+		return;
+	}
+
+#if HAVE_GETIFADDRS
+	ret = getifaddrs(&ifaddrs);
+	for(ifa = ifaddrs; ifa != NULL; ifa = ifa->ifa_next) {
+		if (ifa->ifa_addr == NULL || (ifa->ifa_flags & IFF_LOOPBACK) || (ifa->ifa_addr->sa_family != AF_INET))
+			continue;
```

Comment:
Why do we want to ignore if it is a loopback address? If an application wants to use a particular source address, it should pass it through hints.